scotty/ray_solver.py

```diff
@@ -115,23 +115,53 @@ def event_leave_simulation(
 
     @_event(terminal=True, direction=0.0)
     def event_cross_resonance(tau, ray_parameters_2D, K_zeta, hamiltonian: Hamiltonian):
+        # this is for the fundamental electron cyclotron frequency
         # Currently only works when crossing resonance.
-        # To implement crossing of higher harmonics as well
         # Not implemented for relativistic temperatures
-        delta_gyro_freq = 0.01
+
+        # used to include a delta_gyro_freq where the event triggers when
+        # beam frequency is close to the electron frequency, but this is removed because it works unreliably.
 
         q_R, q_Z, _, _ = ray_parameters_2D
 
         B_R = field.B_R(q_R, q_Z)
         B_T = field.B_T(q_R, q_Z)
         B_Z = field.B_Z(q_R, q_Z)
 
-        B_Total = np.sqrt(B_R**2 + B_T**2 + B_Z**2)
-        gyro_freq = find_normalised_gyro_freq(B_Total, launch_angular_frequency)
+        B_Total = np.sqrt(B_R**2 + B_T**2 + B_Z**2)  # magnitude of B field
+        gyro_freq = find_normalised_gyro_freq(
+            B_Total, launch_angular_frequency
+        )  # find the ratio of beam freq to electron cyclotron freq
+        difference = (
+            gyro_freq - 1
+        )  # find the difference, if the sign changes, it means you have crossed the resonance frequency
```

Comment:
`black` is formatting this oddly because it's trying to keep the comment on the same line. If you put comments on the line before, then `black` will format this a lot more sensibly:

```suggestion
        # find the ratio of beam freq to electron cyclotron freq
        gyro_freq = find_normalised_gyro_freq(B_Total, launch_angular_frequency)
        # find the difference, if the sign changes,
        # it means you have crossed the resonance frequency
        difference = gyro_freq - 1
```

There's another couple places below you might want to move the comments too